### PR TITLE
Add diff tests for collection and database name

### PR DESCRIPTION
**Pull request by chilagrow**
Closes FerretDB/FerretDB#2210.

### CODE REVIEW

**Review by rumyantseva — DISMISSED** [review]
LGTM

**Review by AlekSi — DISMISSED** [review]

**Review by noisersup — APPROVED** [review]

**Review by w84thesun — DISMISSED** [review]

File: tests/diff/names_test.go [hunk 1]
```diff
@@ -28,68 +28,66 @@ func TestDatabaseName(t *testing.T) {
 
 	collectionName := strings.Repeat("a", 10)
 
-	t.Run("ReservedPrefix", func(t *testing.T) {
-		dbName := "_ferretdb_xxx"
-		ctx, db := setup(t)
-		err := db.Client().Database(dbName).CreateCollection(ctx, collectionName)
+	testCases := map[string]string{
```
Comment (chilagrow) [inline tests/diff/names_test.go:31]: Tests each cases mentioned inhttps://github.com/FerretDB/FerretDB/blob/0fcad4dbb26a8ec784fc69ecf312b5f62b99c238/website/docs/diff.md?plain=1#L24

**Comment by b1ron** [comment]
I still seem to be able to create collection names with non-UTF8 characters. Also seen in this [failure](https://github.com/FerretDB/dance/actions/runs/4503337931/jobs/7926395356?pr=374#step:7:25) on CI.

```
branch.txt: main
version.txt: v0.9.4-rc.2-12-gfe97eea3
```

```
> db.createCollection("æ���� ����")
{ "ok" : 1 }
> db.createCollection("Ä°nanÃ§")
{ "ok" : 1 }
> show collections
Ä°nanÃ§
æ���� ����
> 
```

**Comment by chilagrow** [comment]
> I still seem to be able to create collection names with non-UTF8 characters. Also seen in this [failure](https://github.com/FerretDB/dance/actions/runs/4503337931/jobs/7926395356?pr=374#step:7:25) on CI.
> 
> ```
> branch.txt: main
> version.txt: v0.9.4-rc.2-12-gfe97eea3
> ```
> 
> ```
> > db.createCollection("æ���� ����")
> { "ok" : 1 }
> > db.createCollection("Ä°nanÃ§")
> { "ok" : 1 }
> > show collections
> Ä°nanÃ§
> æ���� ����
> > 
> ```

utf8.ValidString is what we are using to check this now. https://github.com/FerretDB/FerretDB/pull/2239#discussion_r1144284737
I updated the test to use invalid utf8 example. Looks like "Ä°nanÃ§" and "æ���� ����" are valid utf8 characters.

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
Nicely done! All the cases are covered, and it's easy to read and understand them.

**Pull request merged**